server/ServerFSM.cpp

```diff
@@ -1273,11 +1273,40 @@ sc::result MPLobby::react(const LobbyUpdate& msg) {
             psd.second.m_save_game_empire_id = ALL_EMPIRES;
         }
 
+        // remove all AIs
+        m_lobby_data->m_players.remove_if([](const std::pair<int, PlayerSetupData>& plr) {
+            return plr.second.m_client_type == Networking::CLIENT_TYPE_AI_PLAYER;
+        });
+        m_ai_next_index = 1;
+
         // refresh save game empire data
         boost::filesystem::path save_dir(GetServerSaveDir());
+        std::vector<PlayerSaveHeaderData> player_save_header_data;
         try {
             LoadEmpireSaveGameData((save_dir / m_lobby_data->m_save_game).string(),
-                                   m_lobby_data->m_save_game_empire_data);
+                                   m_lobby_data->m_save_game_empire_data,
+                                   player_save_header_data);
+
+            for (const auto& pshd : player_save_header_data) {
```

Comment:
add a comment before this block explaining its purpose

Reply:
Added.